homeassistant/helpers/script_variables.py

```diff
@@ -106,19 +108,22 @@ class _ParallelStore:
 class ScriptRunVariables(UserDict[str, Any]):
     """Class to hold script run variables.
 
-    It is a subclass of UserDict, so it has all the usual additional methods available.
-    Operations using those methods are equivalent to assigning values to variables.
+    The purpose of this class is to provide proper variable scoping semantics for scripts.
+    Each instance institutes a new local scope, in which variables can be defined.
+    Each instance has a reference to the previous instance, except for the top-level instance.
+    The instances therefore form a chain, in which variable lookup and assignment is performed.
+    The variables defined lower in the chain naturally override those defined higher up.
     """
 
     # _previous is the previous ScriptRunVariables in the chain
     _previous: ScriptRunVariables | None = None
     # _parent is the previous non-empty ScriptRunVariables in the chain
     _parent: ScriptRunVariables | None = None
 
-    # _local_store is the store for local variables
-    _local_store: dict[str, Any] | None = None
-    # _parallel store is used for each parallel sequence
-    _parallel_store: _ParallelStore | None = None
+    # _local_data is the store for local variables
+    _local_data: dict[str, Any] | None = None
+    # _parallel_data is used for each parallel sequence
```

Comment:
Please elaborate, either here, or preferably in the class docstring, about what's special with variables in parallel sequences and how this class helps with those.

Reply:
Never mind, I think it's already explained above.